utils/db.ts

```diff
@@ -31,75 +30,106 @@ async function getValues<T>(
 }
 
 // Item
-interface InitItem {
+export interface Item {
   userId: string;
   title: string;
   url: string;
-}
-
-export interface Item extends InitItem {
+  // The below properties can be automatically generated upon item creation
   id: string;
   createdAt: Date;
   score: number;
 }
 
-export async function createItem(initItem: InitItem) {
-  const item: Item = {
+export function newItemProps(): Omit<Item, "userId" | "title" | "url"> {
+  return {
     id: crypto.randomUUID(),
     score: 0,
     createdAt: new Date(),
-    ...initItem,
   };
+}
 
-  const itemKey = ["items", item.id];
-  const itemByTime = ["items_by_time", item.createdAt.getTime(), item.id];
+/**
+ * Sets an item in KV.
+ *
+ * @example New item creation
+ * ```ts
+ * import { newItemProps, setItem, incrementAnalyticsMetricPerDay } from "@/utils/db.ts";
+ *
+ * const item: Item = {
+ *   userId: "example-user-id",
+ *   title: "example-title",
+ *   url: "https://example.com"
+ *   ..newItemProps(),
+ * };
+ *
+ * await setItem(item);
+ * await incrementAnalyticsMetricPerDay("items_count", item.createdAt);
+ * ```
+ */
+export async function setItem(item: Item) {
+  const itemsKey = ["items", item.id];
+  const itemsByTimeKey = ["items_by_time", item.createdAt.getTime(), item.id];
   const itemsByUserKey = ["items_by_user", item.userId, item.id];
 
   const res = await kv.atomic()
-    .check({ key: itemKey, versionstamp: null })
-    .check({ key: itemByTime, versionstamp: null })
-    .check({ key: itemsByUserKey, versionstamp: null })
-    .set(itemKey, item)
-    .set(itemByTime, item)
+    .set(itemsKey, item)
+    .set(itemsByTimeKey, item)
     .set(itemsByUserKey, item)
     .commit();
 
   if (!res.ok) throw new Error(`Failed to set item: ${item}`);
+}
 
-  await incrementAnalyticsMetricPerDay("items_count", new Date());
-
-  return item;
-}
-
-export async function getAllItemsInTimeAgo(timeAgo: string) {
-  switch (timeAgo) {
-    case "month":
-      return await getValues<Item>({
-        prefix: ["items_by_time"],
-        start: ["items_by_time", Date.now() - DAY * 30],
-      });
-    case "all":
-      return await getValues<Item>({
-        prefix: ["items_by_time"],
-      });
-    default:
-      return await getValues<Item>({
-        prefix: ["items_by_time"],
-        start: ["items_by_time", Date.now() - WEEK],
-      });
-  }
+export async function deleteItem(item: Item) {
+  const itemsKey = ["items", item.id];
+  const itemsByTimeKey = ["items_by_time", item.createdAt.getTime(), item.id];
+  const itemsByUserKey = ["items_by_user", item.userId, item.id];
+
+  const res = await kv.atomic()
+    .delete(itemsKey)
+    .delete(itemsByTimeKey)
+    .delete(itemsByUserKey)
+    .commit();
+
+  if (!res.ok) throw new Error(`Failed to set item: ${item}`);
 }
 
```

Comment:
Hmm... is the `deleteItem` feature only intended for testing purposes? It could be used after missing `itemsByTimeKey` from the old items is tackled or the entire KV is reset. It is a pretty helpful feature in my opinion.

Reply:
Actually, `deleteItem()` isn't needed in testing or production. So I just removed it.

> It could be used after missing `itemsByTimeKey` from the old items is tackled or the entire KV is reset.

Sorry, what do you mean?

Reply:
```
  const res = await kv.atomic()
    .delete(itemsKey)
    .delete(itemsByTimeKey)
    .delete(itemsByUserKey)
    .commit();
```

For now, there is no `itemsByTimeKey` key-value in production so it might be causing issues when allowing the user to able to delete the item. 

And It is quite a handy feature for users in my opinion.

Ah nvm it is being removed